lib/opam/opam_make_opam.ml

```diff
@@ -0,0 +1,163 @@
+open Core.Std
+open Core_extended.Std
+open Async.Std
+
+let opam_string_of_command_list
+  : String.t List.t -> String.t =
+  fun cmds ->  "[\n" ^
+               (List.fold ~init:""
+                  ~f:(fun acc el -> acc ^ "    [" ^ el ^ "]\n") cmds) ^
+               "]\n"
+let format_cmds
+  : default:String.t List.t -> String.t List.t -> String.t =
+  fun ~default cmds ->
+    match cmds with
+    | [] -> opam_string_of_command_list default
+    | _ -> opam_string_of_command_list cmds
+
+let create_build_cmds
```

Comment:
I am not sure I am suggesting "sticking the format commands into main function". Do you mean calling `format_cmds` directly from `write_opam`?

If that is the case, it is not necessarily the only solution to reduce duplication. I would go with something like `default_to_make target -> format_cmds ~default:["make "  ^ target]` or something of the like and keep the specific functions, but calling that one. But feel free to drop this if you think is excessive, not really an issue